src/main/java/org/terasology/lost/OnSpawnSystem.java

```diff
@@ -15,26 +15,173 @@
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+
+import java.util.Set;
 
 @RegisterSystem(RegisterMode.AUTHORITY)
 public class OnSpawnSystem extends BaseComponentSystem {
     @In
     private EntityManager entityManager;
     @In
     private InventoryManager inventoryManager;
+    @In
+    private Console console;
+    @In
+    private AssetManager assetManager;
+    @In
+    private WorldProvider worldProvider;
 
     @ReceiveEvent
     public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player, InventoryComponent inventory) {
         inventoryManager.giveItem(player, null, entityManager.create("Lost:antrumSabre"));
+        ProgressTrackingComponent progressTrackingComponent = new ProgressTrackingComponent();
+        progressTrackingComponent.addLevel("Grassland", "Lost:stonehengeWipedOutKey");
+        progressTrackingComponent.addLevel("Tundra", "Lost:pyramidBladesOfTruth");
+        progressTrackingComponent.addLevel("Bare", "Lost:pyramidBladesOfTruth");
+        progressTrackingComponent.addLevel("Scorched", "Lost:pyramidBladesOfTruth");
+        progressTrackingComponent.addLevel("Temperate desert", "Lost:pyramidBladesOfTruth");
+        progressTrackingComponent.addLevel("Temperate rain forest", "Lost:templePlasmaOfFire");
+        progressTrackingComponent.addLevel("Temperate deciduous forest", "Lost:templePlasmaOfFire");
+        progressTrackingComponent.addLevel("Subtropical desert", "Lost:pyramidBladesOfTruth");
+        progressTrackingComponent.addLevel("Shrubland", "Lost:stonehengeWipedOutKey");
+        progressTrackingComponent.addLevel("Marsh", "Lost:stonehengeWipedOutKey");
+        progressTrackingComponent.addLevel("Tropical rain forest", "Lost:templePlasmaOfFire");
+        progressTrackingComponent.addLevel("Tropical seasonal forest", "Lost:templePlasmaOfFire");
+        progressTrackingComponent.addLevel("Lakeshore", "Lost:well");
+        progressTrackingComponent.addLevel("Coast", "Lost:well");
+        progressTrackingComponent.addLevel("Beach", "Lost:well");
+        player.addComponent(progressTrackingComponent);
+
+        LocationComponent loc = player.getComponent(LocationComponent.class);
+        Vector3f playerLocation = loc.getWorldPosition();
+        Prefab prefab =
+                assetManager.getAsset("Lost:hut", Prefab.class).orElse(null);
+        EntityBuilder entityBuilder = entityManager.newBuilder(prefab);
+        EntityRef item = entityBuilder.build();
+        int x = (int) Math.round(playerLocation.getX()) + 5;
+        int y = (int) Math.round(playerLocation.getZ()) + 5;
+        Vector3i spawnPosition = new Vector3i(x, getGroundHeight(x, y), y);
+        BlockRegionTransform b = BlockRegionTransform.createRotationThenMovement(Side.FRONT, Side.FRONT,
+                spawnPosition);
+        item.send(new SpawnStructureEvent(b));
+    }
+
+    @ReceiveEvent
+    public void onBiomeChange(OnBiomeChangedEvent event, EntityRef player) {
```

Comment:
I don't think this should live in the `OnSpawnSystem`.

Reply:
Moved to a new class LevelSpawnSystem